src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -1513,12 +1526,22 @@ protected override void HandleSetVariableRequestAsync(IRequestResponder<SetVaria
         }
 
         /// <summary>
-        /// Currently unsupported. This message can be received when we return a source file that doesn't exist (such as a library within gdb).
-        /// See github issue: microsoft/vscode-cpptools#3662
+        /// This message will only be received if a Source uses a sourceReference instead of filename
+        /// So use it for non-existing sources
         /// </summary>
         protected override void HandleSourceRequestAsync(IRequestResponder<SourceArguments, SourceResponse> responder)
         {
-            responder.SetError(new ProtocolException("'SourceRequest' not supported."));
+            var source = responder.Arguments.Source;
+            var response = new SourceResponse();
+            if (source != null)
+            {
+                int sourceRef = (int)source.SourceReference;
+                string content = TextPositionTuple.GetSourceForRef(sourceRef);
+                response.MimeType = "text/x-lldb.disassembly";
```

Comment:
>x-lldb [](start = 42, length = 6)

Does something in VS Code already understand this type?

Reply:
It's borrowed from the CodeLLDB extension.
Asm CodeLens doesn't support this kind of disassembly yet.

Reply:
I don't know what VS Code uses the MimeType for. Does it matter what we put? Does it matter if CodeLLDB isn't installed?

Reply:
If it isn't installed you won't get syntax highlighting. If it is you will.